lib/ansible/modules/network/vyos/vyos_interface.py

```diff
@@ -227,7 +227,7 @@ def map_obj_to_commands(updates):
 
 
 def map_config_to_obj(module):
-    data = get_config(module)
+    data = get_config(module, flags=['| grep interface'])
```

Comment:
Maybe I am confused about this patch, but this is going to change functionality of the vyos_interface module now right? Is that what we want here or is this testing that nothing is broken?

What I am envisioning is we update the test in test/integration/targets/vyos_interface to test both cases where we pass flags and don't pass flags. It ensure we keep old behaviour and test new.

Reply:
> Maybe I am confused about this patch, but this is going to change functionality of the vyos_interface module now right? Is tha t what we want here or is this testing that nothing is broken?
> 
> What I am envisioning is we update the test in test/integration/targets/vyos_interface to test both cases where we pass flags and don't pass flags. It ensure we keep old behaviour and test new.

@pabelanger  So  earlier for vyos_interface module without flags we were fetching all the running configuration from the device instead of only having interface related config , but with this flag only interfaces related config will be fetched from device. So with current test cases we have tested this and nothing seems to be broken . So basically I have observed testing for following  scenarios :
1. When flag is not passed : For other vyos modules no flag have been passed, and all the test cases are passing .
2. When flag is passed: At present for vyos_interface we are passing flag, and all test cases passing successfully for this change . 
        I hope I am able to clarify things now :)  .  
  
@NilashishC  whats your take on this  ?

Reply:
@rohitthakur2590 LGTM. With the ability to pass in resource specific filters while fetching running-config from a device, we will be able to reduce amount of config lines transferred over the wire. However, there's no way we can test this in the integration tests directly since this concerns the module performance and not functionality. Your way of asserting that this works fine seems good to me.